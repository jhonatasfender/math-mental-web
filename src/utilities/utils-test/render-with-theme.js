import Main from '@components/layout/main';
import { render } from '@testing-library/react';

export const renderWithMain = (children) => render(<Main>{children}</Main>);
