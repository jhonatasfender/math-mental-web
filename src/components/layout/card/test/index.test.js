import Card from '@components/layout/card';
import { renderWithMain } from '@utils-test/render-with-theme';
import { ThemeProvider } from 'styled-components';

describe('<Card />', () => {
  it('should render', () => {
    const { container } = renderWithMain(
      <ThemeProvider theme={{ sector: 'addition' }}>
        <Card />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
