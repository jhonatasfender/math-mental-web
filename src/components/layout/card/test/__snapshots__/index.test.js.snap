// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render 1`] = `
.c7 {
  position: absolute;
  top: 7rem;
  left: 0;
  width: 13.2rem;
}

.c9 {
  width: 100%;
  background: #e8e6e5;
  height: 0.7rem;
  border-radius: 1rem;
  margin: 0 1rem;
}

.c10 {
  background: #2565D2;
  width: 30%;
  height: 100%;
  border-radius: 1rem;
  border: none;
}

.c5 {
  width: auto;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c4 {
  border-radius: 1rem;
  min-width: 13rem;
  width: 13rem;
  height: 13rem;
  box-shadow: 5px 0px 9px 0px #AAF2EA,-9px 0px 17px -3px #AAF2EA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  overflow: hidden;
  color: #CCF8EA;
  margin: 1rem;
  background: #072ac8;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  -webkit-transition: -webkit-transform 1s;
  -webkit-transition: transform 1s;
  transition: transform 1s;
}

.c4 > span {
  font-size: 0.9rem;
  font-weight: 100;
}

.c8 {
  width: 100%;
  background: #F9F9F9;
  height: 3rem;
  position: absolute;
  bottom: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-items: center;
  z-index: 1;
}

.c6 {
  font-size: 2rem;
  height: 8rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  width: 3rem;
}

.c3 {
  padding: 1rem 17rem;
  width: 100wh;
  background: #EDEDED;
}

.c3 * {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
}

<div>
  <header
    class="c0"
  >
    <svg
      aria-hidden="false"
      aria-label="dark mode"
      class="c1 c2"
      fill="currentColor"
      focusable="false"
      role="button"
      type="button"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 22a10 10 0 1 0 0-20 10 10 0 0 0 0 20zm0-2V4a8 8 0 1 1 0 16z"
      />
    </svg>
  </header>
  <div
    class="c3"
  >
    <div
      aria-label="level 1"
      class="c4"
    >
      <span>
        level 
        1
      </span>
      <div
        class="c5 c6"
      />
      <svg
        class="c7"
        viewBox="0 0 1440 400"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M 0,400 C 0,400 0,100 0,100 C 50.28840307700442,114.51226572206508 100.57680615400884,129.02453144413016 148,134 C 195.42319384599116,138.97546855586984 239.9811784609691,134.41413994554443 288,120 C 336.0188215390309,105.58586005445557 387.4984800021148,81.31890877369213 434,70 C 480.5015199978852,58.68109122630788 522.0249015305718,60.310224959687034 576,79 C 629.9750984694282,97.68977504031297 696.4019138755981,133.44019138755976 746,129 C 795.5980861244019,124.55980861244022 828.3674429670357,79.92900949007377 867,74 C 905.6325570329643,68.07099050992623 950.1283142562586,100.84377065214517 1003,106 C 1055.8716857437414,111.15622934785483 1117.1193000079304,88.69590790134552 1160,76 C 1202.8806999920696,63.30409209865447 1227.3944857120198,60.372597742472706 1271,66 C 1314.6055142879802,71.6274022575273 1377.3027571439902,85.81370112876365 1440,100 C 1440,100 1440,400 1440,400 Z"
          fill="#4498DB"
          stroke="none"
          stroke-width="0"
        />
        <path
          d="M 0,400 C 0,400 0,200 0,200 C 46.46767030585001,211.15587247878614 92.93534061170001,222.31174495757224 147,232 C 201.0646593883,241.68825504242776 262.72630785904994,249.9088926484972 314,235 C 365.27369214095006,220.0911073515028 406.1594279521003,182.05268444843904 451,181 C 495.8405720478997,179.94731555156096 544.635980332549,215.88036955774672 587,225 C 629.364019667451,234.11963044225328 665.2966507177034,216.42583732057417 710,218 C 754.7033492822966,219.57416267942583 808.1774167966373,240.41628115995664 859,232 C 909.8225832033627,223.58371884004336 957.9936820957469,185.9090380395993 1006,184 C 1054.0063179042531,182.0909619604007 1101.8478548203757,215.94756668164638 1147,230 C 1192.1521451796243,244.05243331835362 1234.6148986227497,238.3006952338153 1283,230 C 1331.3851013772503,221.6993047661847 1385.692550688625,210.84965238309235 1440,200 C 1440,200 1440,400 1440,400 Z"
          fill="#88E4EB"
          stroke="none"
          stroke-width="0"
        />
        <path
          d="M 0,400 C 0,400 0,300 0,300 C 47.0964868222792,322.0315366517751 94.1929736445584,344.0630733035502 147,334 C 199.8070263554416,323.9369266964498 258.32459224404556,281.77924343757434 308,278 C 357.67540775595444,274.22075656242566 398.5086573792593,308.8199529461524 441,317 C 483.4913426207407,325.1800470538476 527.6407782389172,306.94094477781607 570,305 C 612.3592217610828,303.05905522218393 652.9282296650719,317.41626794258366 707,312 C 761.0717703349281,306.58373205741634 828.6463031007955,281.3939834518492 881,282 C 933.3536968992045,282.6060165481508 970.4865579317454,309.00779825001985 1016,304 C 1061.5134420682546,298.99220174998015 1115.4074651722224,262.57482354807155 1165,268 C 1214.5925348277776,273.42517645192845 1259.883581379365,320.69290755769384 1305,333 C 1350.116418620635,345.30709244230616 1395.0582093103176,322.6535462211531 1440,300 C 1440,300 1440,400 1440,400 Z"
          fill="#F9F9F9"
          stroke="none"
          stroke-width="0"
        />
      </svg>
      <div
        class="c8"
      >
        <div
          class="c9"
        >
          <div
            class="c10"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
